src/main/scala/chiseltest/package.scala

```diff
@@ -37,15 +82,15 @@ package object chiseltest {
       case (x: Interval, value: Interval) =>
         require(x.binaryPoint == value.binaryPoint, "binary point mismatch")
         pokeBits(x, value.litValue)
-      case (x: Bundle, value: Bundle) => {
+      case (x: Record, value: Record) => {
         // TODO: chisel needs to expose typeEquivalent
-        require(x.elements.keys == value.elements.keys)  // TODO: this discards type data =(
+        DataMirror.checkTypeEquivalence(x, value)
```

Comment:
checkTypeEquivalence returns a boolean value, right? So this just gets a boolean and discards it, instead of asserting on it?